src/main/java/com/letsTravel/LetsTravel/controller/map/CountryController.java

```diff
@@ -14,18 +19,22 @@
 @RequestMapping("/api/countries")
 public class CountryController {
 
-    private final CountryService countryService;
+	private final CountryService countryService;
 
-    @GetMapping("/list")
-    public ResponseEntity<List<Country>> getAllCountries() {
-        return ResponseEntity.ok(countryService.getAllCountries());
-    }
+	@GetMapping("/list")
+	public ResponseEntity<List<Country>> getAllCountries(
+			@RequestParam(value = "page") int page,
+			@RequestParam(value = "size") int size,
+			@RequestParam(value = "orderby") String orderBy,
+			@RequestParam(value = "sort") String sort
+			) {
+		Pageable pageable = sort.equals("DESC") ? PageRequest.of(page, size, Sort.by(Sort.Direction.DESC, orderBy)): PageRequest.of(page, size, Sort.by(Sort.Direction.ASC, orderBy));
```

Comment:
```suggestion
Pageable pageable = PageRequest.of(pageDto.getPage(), pageDto.getSize(), Sort.by(Sort.Direction.fromString(pageDto.getSort()), pageDto.getOrderBy()));
```

Reply:
이렇게 동적으로 하는게 나을 듯 함
그리고 로직은 service 단으로 이동하는게 나을 듯 함

Reply:
만약 오름, 내림 말고 다른 정렬기준이 생기면 해당 문자열에 맞는 Sort 객체 반환하는 메소드 하나 추가해서 바꾸면 될 듯 fromString을 그 메소드로 바꾸면 될듯

Reply:
public class SortUtils {

    public static Sort getSortBasedOnCustomDirection(String direction, String orderBy) {
        switch (direction) {
            case "ASC":
                return Sort.by(Sort.Direction.ASC, orderBy);
            case "DESC":
                return Sort.by(Sort.Direction.DESC, orderBy);
            case "ALPHABETICAL":
                return Sort.by(Sort.Direction.ASC, orderBy);  // 알파벳 순으로 정렬
            case "NUMERICAL":
                return Sort.by(Sort.Direction.ASC, orderBy);  // 숫자 순으로 정렬
            default:
                return Sort.by(Sort.Direction.DESC, orderBy);  // 기본값은 DESC
        }
    }
}
대충 이런거 추가하고,
```suggestion
                Pageable pageable = PageRequest.of(pageDto.getPage(), pageDto.getSize(), 
                SortUtils.getSortBasedOnCustomDirection(pageDto.getSort(), pageDto.getOrderBy()));
```

이렇게?

Reply:
참고로 
 case "ALPHABETICAL":
            return Sort.by(Sort.Direction.ASC, orderBy);  // 알파벳 순으로 정렬
        case "NUMERICAL":
            return Sort.by(Sort.Direction.ASC, orderBy);  // 숫자 순으로 정렬
   는 from gpt로 나는처음보는 정렬방식